### PR TITLE
Pause OS upgrade when coinciding with Vespa upgrade

### PR DESCRIPTION
**Pull request by mpolden**

### CODE REVIEW

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/os/OsVersions.java
Comment (freva) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/os/OsVersions.java]: Copy active state from `oldTarget` if present?

**Review by freva — COMMENTED** [review]
I thought the main issue is that the children cant upgrade? This PR is effectively no-op as host-admin will always converge towards `SUSPENDED_PROCESS` (f.ex. to upgrade host-admin) ahead of `SUSPENDED_HOST` (f.ex. OS upgrade or reboot).

But if we extend `OsUpgradeActivator::canUpgradeOsOf` to consider children, then this will almost never be active? I think this logic should just be moved to `YLinuxUpgradeTask`: skip the task if any of its children are upgrading.

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/os/OsVersions.java
Comment (mpolden) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/os/OsVersions.java]: Setting the same version is a no-op (checked earlier in this method), but I can add a test for this case.

**Comment by mpolden** [comment]
> I thought the main issue is that the children cant upgrade?

No, the idea is to pause OS upgrade while _infrastructure_  is upgraded. Convergence of infrastructure during release typically times out if an OS upgrade is ongoing.

The workaround we used to unblock releases was to set `wantedOsVersion` to `null`, which allows nodes currently upgrading to complete, but not start any new upgrades. This change does essentially the same thing.

I assume OS upgrades will take way too long to complete if we pause on application upgrades as well.

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/os/OsVersions.java
Comment (freva) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/os/OsVersions.java]: I meant to copy the active state, not the version? I.e. if we were upgrading to X and it was `active`, then upgrading to `y` should also be `active`? No need to set to false and wait for the next iteration of the maintainer to set back to `active`?

**Comment by freva** [comment]
Ok, I remember it now; host-admin individually correctly prioritizes to upgrade it self before upgrading the OS, but the issue was that hosts that have node(s) of one of the big applications (many nodes, many docs, and/or lots of feeding) will take long due to resume post OS upgrade due to waiting for their _node_ to come up. The _node_ takes time due to f.ex. redistribution of new documents, fed while it was down.

In that case, this PR will indeed fix the problem.

**Review by freva — APPROVED** [review]

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/os/OsVersions.java
Comment (freva) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/os/OsVersions.java]:
```suggestion
osVersions.put(nodeType, new OsVersion(newTarget, oldTarget.map(OsVersion::active).orElse(false)));
```

Comment (mpolden) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/os/OsVersions.java]: True, I realised that posting my comment. We can do that, although I don't think it matters much in practice.

Comment (mpolden) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/os/OsVersions.java]: Actually, let's leave it as-is. I think it's more clean to let `OsUpgradeActivator` be the sole thing that activates versions (as stated in javadoc).

**Comment by mpolden** [comment]
.

**Pull request merged**
